From 2dbea6690bb772b7896071d61c965f81ffbe1445 Mon Sep 17 00:00:00 2001
From: Kevin Hao <kexin.hao@windriver.com>
Date: Tue, 26 Nov 2019 18:00:15 +0800
Subject: [PATCH 400/767] net: octeontx2-pf: Don't set the parse mode for LBKs

The LBKs are not allowed to set the parse mode. It will return
an error if we try to do so, then cause a failure when opening
these ports.

Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
index a425ebcbd4b8..d40ca122a4a9 100644
--- a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
+++ b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
@@ -1648,10 +1648,12 @@ int otx2_open(struct net_device *netdev)
 		otx2_config_hw_rx_tstamp(pf, true);
 	}
 
-	/* Set NPC parsing mode */
-	err = otx2_set_npc_parse_mode(pf);
-	if (err)
-		goto err_free_cints;
+	/* Set NPC parsing mode, skip LBKs */
+	if (!is_otx2_lbkvf(pf->pdev)) {
+		err = otx2_set_npc_parse_mode(pf);
+		if (err)
+			goto err_free_cints;
+	}
 
 	err = otx2_rxtx_enable(pf, true);
 	if (err)
-- 
2.31.1

